nagios/status_test.go

```diff
@@ -0,0 +1,542 @@
+				Critical: true,
+				Unknown:  true,
+				Pending:  true,
+			},
+			want: "ok warning critical unknown pending",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.serviceStatus.String(); got != tt.want {
+				t.Errorf("String() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGeneralServiceRequest_build(t *testing.T) {
+	type args struct {
+		query             string
+		includeStartCount bool
+	}
+	tests := []struct {
+		name                  string
+		generalServiceRequest GeneralServiceRequest
+		args                  args
+		want                  Query
+	}{
+		{
+			name:                  "blank",
+			generalServiceRequest: GeneralServiceRequest{},
+			args: args{
+				query:             "",
+				includeStartCount: false,
+			},
+			want: Query{
+				Endpoint: statusEndpoint,
+				URLQuery: url.Values{},
+			},
+		},
+		{
+			name:                  "blank with Start and Count",
+			generalServiceRequest: GeneralServiceRequest{Count: 1},
+			args: args{
+				query:             "",
+				includeStartCount: true,
+			},
+			want: Query{
+				Endpoint: statusEndpoint,
+				URLQuery: url.Values{
+					"start": []string{"0"},
+					"count": []string{"1"},
+				},
+			},
+		},
+		{
+			name:                  "blank with EndTime",
+			generalServiceRequest: GeneralServiceRequest{EndTime: 1},
+			args: args{
+				query:             "",
+				includeStartCount: false,
+			},
+			want: Query{
+				Endpoint: statusEndpoint,
+				URLQuery: url.Values{
+					"starttime": []string{"0"},
+					"endtime":   []string{"1"},
+				},
+			},
+		},
+		{
+			name:                  "blank with ShowDetails",
+			generalServiceRequest: GeneralServiceRequest{ShowDetails: true},
+			args: args{
+				query:             "",
+				includeStartCount: false,
+			},
+			want: Query{
+				Endpoint: statusEndpoint,
+				URLQuery: url.Values{
+					"details": []string{"true"},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.generalServiceRequest.build(tt.args.query, tt.args.includeStartCount); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("build() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestServiceCountRequest_Build(t *testing.T) {
+	tests := []struct {
+		name                  string
+		GeneralServiceRequest GeneralServiceRequest
+		want                  Query
+	}{
+		{
+			name:                  "blank",
+			GeneralServiceRequest: GeneralServiceRequest{},
+			want: Query{
+				Endpoint: statusEndpoint,
+				URLQuery: url.Values{
+					"query": []string{"servicecount"},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			a := ServiceCountRequest{
+				GeneralServiceRequest: tt.GeneralServiceRequest,
+			}
+			if got := a.Build(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Build() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestServieListRequest_Build(t *testing.T) {
```

Comment:
```suggestion
func TestServiceListRequest_Build(t *testing.T) {
```

BTW you can generate these test if you're using some kind of IDE (although cleaning them up is a bit of work).